docs/server-api-auth.md

```diff
@@ -0,0 +1,145 @@
+---
+id: server-api-auth
+title: Set up server API authorization
+sidebar_label: API auth setup
```

Comment:
i think the "job to be done" of this page is just security - we shouldnt limit ourselves to just the api authz feature. look at how other projects do it:

- https://grafana.com/docs/grafana/latest/administration/security/
- https://www.cockroachlabs.com/docs/stable/security-overview.html

think from the POV of the customer?

```suggestion
id: server-security
title: Temporal Server Security
sidebar_label: Security
```

Reply:
+1

Reply:
I think server-security needs to start from a higher level and include TLS as a subsection before we get to authZ.

Reply:
I grabbed Sergey's draft on Temporal Server OSS security from notion and used that as the basis of context in order to do this -